### PR TITLE
Update for ITK 5.2

### PR DESCRIPTION
**Pull request by tbirdso**
- Update package requirements for ITK v5.2 release
- Update for PyVectorContainer template changes introduced in ITK 5.2 (this affects geometry rendering)
- Drop CI checks for Python 3.5 which is no longer maintained

Resolves package dependency issues noted in #406.

### CODE REVIEW

**Comment by tbirdso** [comment]
I think failures stem from an issue with an outdated `itk-meshtopolydata` package -- seeing an `ImportError` resulting from changes that are present in ITK 5.2:
>  from itk.support import itkHelpers
E   ImportError: cannot import name 'itkHelpers' from 'itk.support' (unknown location)

@thewtex Given that checks are passing, do we need to publish an updated `itk-meshtopolydata` package for 5.2 compatibility?

**Comment by tbirdso** [comment]
Observed `PyVectorContainer` failures are due to template changes for the `itk::PyVectorContainer` class introduced in ITK 5.2.0 to align more closely with the `itk::VectorContainer` class. See [PR 408](https://github.com/InsightSoftwareConsortium/itkwidgets/pull/408).

**Comment by tbirdso** [comment]
Note integration failures predate these changes -- see [Run 2099157846](https://github.com/InsightSoftwareConsortium/itkwidgets/runs/2099157846)

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       max-parallel: 4
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python-version: [3.5, 3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8]
```
Comment (thewtex) [inline .github/workflows/build-test.yml:13]: Can we add 3.9 here?

File: itkwidgets/_transform_types.py [hunk 2]
```diff
@@ -313,13 +313,23 @@ def to_point_set(point_set_like):  # noqa: C901
...
         element_type = point_template[1][1]
+        
         # todo: test array_view here and below
-        point_values = itk.PyVectorContainer[element_type].array_from_vector_container(
-            points)
-        if len(
-                point_values.shape) > 1 and point_values.shape[1] == 2 or point_values.shape[1] == 3:
+        # Make use of functional interface if available
+        if hasattr(itk, 'array_from_vector_container'):
+            point_values = itk.array_from_vector_container(points)
```
Comment (thewtex) [inline itkwidgets/_transform_types.py:323]: Nice :-)

File: itkwidgets/_transform_types.py [hunk 3]
```diff
@@ -313,13 +313,23 @@ def to_point_set(point_set_like):  # noqa: C901
...
+        else:
+            # Accommodate PyVectorContainer legacy wrapping for element type only
```
Comment (thewtex) [inline itkwidgets/_transform_types.py:325]: Let's only support the latest version.

File: itkwidgets/_transform_types.py [hunk 4]
```diff
@@ -334,10 +344,21 @@ def to_point_set(point_set_like):  # noqa: C901
...
-            data = itk.PyVectorContainer[pixel_type].array_from_vector_container(
-                itk_point_data)
+            point_data_template = itk.template(itk_point_data)
+            element_type = point_data_template[1][1]
+
+            # Make use of functional interface if available
+            if hasattr(itk, 'array_from_vector_container'):
+                data = itk.array_from_vector_container(itk_point_data)
+            else:
+                # Accommodate PyVectorContainer legacy wrapping for element type only
```
Comment (thewtex) [inline itkwidgets/_transform_types.py:354]: Remove

File: itkwidgets/_transform_types.py [hunk 5]
```diff
@@ -423,13 +444,23 @@ def to_geometry(geometry_like):  # noqa: C901
...
         element_type = point_template[1][1]
+        
         # todo: test array_view here and below
-        point_values = itk.PyVectorContainer[element_type].array_from_vector_container(
-            points)
+        # Make use of functional interface if available
+        if hasattr(itk, 'array_from_vector_container'):
+            point_values = itk.array_from_vector_container(points)
+        else:
+            # Accommodate PyVectorContainer legacy wrapping for element type only
```
Comment (thewtex) [inline itkwidgets/_transform_types.py:457]: Remove

File: itkwidgets/_transform_types.py [hunk 6]
```diff
@@ -451,22 +482,40 @@ def to_geometry(geometry_like):  # noqa: C901
         for cell_type, itk_cells in [('verts', itk_verts), ('lines', itk_lines),
                                      ('polys', itk_polys), ('strips', itk_strips)]:
             if itk_cells.Size():
-                data = itk.PyVectorContainer[itk.UI].array_from_vector_container(
-                    itk_cells)
+                # Make use of functional interface if available
+                if hasattr(itk, 'array_from_vector_container'):
+                    data = itk.array_from_vector_container(itk_cells)
+                else:
```
Comment (thewtex) [inline itkwidgets/_transform_types.py:488]: Remove

File: itkwidgets/_transform_types.py [hunk 7]
```diff
@@ -483,10 +532,21 @@ def to_geometry(geometry_like):  # noqa: C901
...
-            data_type, number_of_components = _itk_pixel_to_vtkjs_type_components[pixel_type]
-            data = itk.PyVectorContainer[pixel_type].array_from_vector_container(
-                itk_cell_data)
+            point_data_template = itk.template(itk_point_data)
+            element_type = point_data_template[1][1]
+
+            # Make use of functional interface if available
+            if hasattr(itk, 'array_from_vector_container'):
+                data = itk.array_from_vector_container(itk_cell_data)
+            else:
```
Comment (thewtex) [inline itkwidgets/_transform_types.py:541]: Remove

**Review by thewtex — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (tbirdso) [reply .github/workflows/build-test.yml]: Added.

**Review by thewtex — APPROVED** [review]
@tbirdso thanks!

**Pull request merged**